xclim/core/units.py

```diff
@@ -349,10 +349,8 @@ def convert_units_to(  # noqa: C901
                         # The new cf standard name is inserted by the converter
                         try:
                             source = _CONVERSIONS[(convname, direction)](source)
-                        except Exception:
-                            # FIXME: This is a broad exception. Bad practice.
-                            # Failing automatic conversion
-                            # It will anyway fail further down with a correct error message.
+                        except KeyError:  # noqa: S110
+                            # There is no conversion available for this standard name
```

Comment:
The previous problematic `except Exception` also caught any exception that were raised from the call of the conversion function. Values of `_CONVERSIONS` are functions like `rate2amount` and family.

If the conversion failed, then `source_unit` stays the original units and at line 367 (365 in this PR), the conversion fails with an error message given by pint. That was the objective of having a broad `except` : let pint print the error.

I agree that this is weird and not explicit at all. The solution might be to avoid the KeyError:
```python3
if (convname, direction) in _CONVERSIONS:
    try:
        source = _CONVERSIONS[(convname, direction)](source)
    except Exception as err:
        raise ValueError("some intelligent message") from err
    else:
        source_unit = units2pint(source)
```